tests/lint/createJunitReport.py

```diff
@@ -30,7 +30,7 @@
 
 
 def makeJunitXML(inFileName, outFileName):
-	with open(inFileName, 'rt', encoding='UTF-8') as flake8In:
+	with open(inFileName, 'rt', encoding='ANSI', errors='replace') as flake8In:

```

Comment:
@feerrenrut
Do you think we might reconsider this line now that Flake8 runs in UTF-8 Mode as of #11108?
`errors='replace'` is probably still a good strategy, though.

Reply:
Perhaps so, I vaguely remember that this was a pain. If I recall, the encoding of the file was for some reason different on appveyor than it was locally.

Reply:
> the encoding of the file was in the default encoding of the system

Yep, that sounds familiar.

> My best bet is that we should probably add the `-Xutf8` to most if not all invocation of `py`

This sounds like a reasonable approach. Is there anyway we can just set it a default option for python?

Reply:
> This sounds like a reasonable approach. Is there anyway we can just set it a default option for python?

Fortunately yes, there is one.
[PEP 540](https://www.python.org/dev/peps/pep-0540/) introduced the use of the `PYTHONUTF8` environment variable. When set to `1`, UTF-8 Mode is activated.
Note however that `PYTHONIOENCODING` and `PYTHONLEGACYWINDOWSFSENCODING` ([PEP 529](https://www.python.org/dev/peps/pep-0529/#add-legacy-mode)) take precedence when set.
If this approach is fine for AppVeyor build, I guess my recent addition of `-Xutf8` for Flake8 shall then be removed from `appveyor.xml`

Reply:
Hmm, I think I prefer the `-Xutf8` approach over environment variables, I might be paranoid but I feel it's very easy for environment variables to be modified by some layer and become really hard to find the cause of. We should probably be using something like [tox](https://tox.readthedocs.io/en/latest/)